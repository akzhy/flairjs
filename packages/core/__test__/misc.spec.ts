import { describe, expect, test } from 'vitest'

import { readFileSync } from 'node:fs'
import path from 'node:path'
import { dirname } from 'path'
import { fileURLToPath } from 'url'
import { transformCode } from '../index'

const __filename = fileURLToPath(import.meta.url)
const __dirname = dirname(__filename)

const conflictingClassNamesContent = readFileSync(path.resolve(__dirname, './snippets/misc-conflicting-classnames.tsx'), 'utf-8')
const combinedStylesContent = readFileSync(path.resolve(__dirname, './snippets/misc-combined-styles.tsx'), 'utf-8')

describe('Misc tests', () => {
  test('conflicting classnames in same file are working', () => {
    const regex = new RegExp("^class[A-Z][A-Za-z0-9_]*");
    const result = transformCode(conflictingClassNamesContent, 'index.tsx', {
      classNameList: ['className', 'containerClassName', regex.toString()],
      cssOutDir: path.resolve(__dirname, './.css'),
    })
    if (!result) {
      throw new Error('transformCode returned null or undefined')
    }
    expect(result.code).toMatchSnapshot()
  })
  test('combined styles are working', () => {
    const regex = new RegExp("^class[A-Z][A-Za-z0-9_]*");
    const result = transformCode(combinedStylesContent, 'index.tsx', {
      classNameList: ['className', 'containerClassName', regex.toString()],
      cssOutDir: path.resolve(__dirname, './.css'),
    })
    if (!result) {
      throw new Error('transformCode returned null or undefined')
    }
    expect(result.code).toMatchSnapshot()
  })
})
