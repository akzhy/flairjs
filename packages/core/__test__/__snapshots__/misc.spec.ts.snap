// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Misc tests > class component flair is working 1`] = `
"import "@flairjs/client/generated-css/de3088a18b2aa97c.css";
import { flair } from "@flairjs/client";
import clsx from "clsx";
import { Component } from "react";
class Box extends Component<{
	children?: React.ReactNode;
	containerClassName?: string;
}> {
	render() {
		const { containerClassName, children } = this.props;
		return <div className={clsx("QKQ_Zq_box", containerClassName)}>
        {children}
        <span className="title">Hello</span>
      </div>;
	}
}
"
`;

exports[`Misc tests > class component style tags are working 1`] = `
"import "@flairjs/client/generated-css/cbd93da89790ac1c.css";
import { Style } from "@flairjs/client/react";
import clsx from "clsx";
import { Component } from "react";
class Box extends Component<{
	children?: React.ReactNode;
	containerClassName?: string;
}> {
	render() {
		const { children, containerClassName } = this.props;
		return <div className={clsx("ubLe5a_box", containerClassName)}>
        {children}
        <span className="title">Hello</span>
        
      </div>;
	}
}
export { Box };
"
`;

exports[`Misc tests > combined styles are working 1`] = `
"import "@flairjs/client/generated-css/1e01b777eaf6a095.css";
import clsx from "clsx";
import { flair } from "@flairjs/client";
import { Style } from "@flairjs/client/react";
const Box = ({ children, containerClassName }: {
	children?: React.ReactNode;
	containerClassName?: string;
}) => {
	return <div className={clsx("MbGWrq_box", containerClassName)}>
      {children}
      <span className="MbGWrq_title">Hello</span>
      
    </div>;
};
const SecondBox = ({ children, classList }: {
	children?: React.ReactNode;
	classList?: string[];
}) => {
	return <div className={clsx("_8ETg1a_box", classList)}>{children}</div>;
};
"
`;

exports[`Misc tests > conflicting classnames in same file are working 1`] = `
"import "@flairjs/client/generated-css/8f9c0925524c26c8.css";
import clsx from "clsx";
import { flair } from "@flairjs/client";
const Box = ({ children, containerClassName }: {
	children?: React.ReactNode;
	containerClassName?: string;
}) => {
	return <div className={clsx("pkTJfG_box", containerClassName)}>{children}</div>;
};
const SecondBox = ({ children, classList }: {
	children?: React.ReactNode;
	classList?: string[];
}) => {
	return <div className={clsx("-oeEAW_box", classList)}>{children}</div>;
};
"
`;

exports[`Misc tests > different function variants are working 1`] = `
"import "@flairjs/client/generated-css/cc835155a66ce982.css";
import { flair } from "@flairjs/client";
import { Style } from "@flairjs/client/react";
import clsx from "clsx";
import { forwardRef } from "react";
export const Box = ({ children, containerClassName }: {
	children?: React.ReactNode;
	containerClassName?: string;
}) => {
	return <div className={clsx("GCVsQq_box", containerClassName)}>
      {children}
      <span className="title">Hello</span>
    </div>;
};
export function Box2({ children, containerClassName }: {
	children?: React.ReactNode;
	containerClassName?: string;
}) {
	return <div className={clsx("emcImW_box", containerClassName)}>
      {children}
      <span className="title">Hello</span>
    </div>;
}
export const Box3: React.FC<{
	children?: React.ReactNode;
	containerClassName?: string;
}> = ({ children, containerClassName }) => {
	return <div className={clsx("zQJkaa_box", containerClassName)}>
      {children}
      <span className="title">Hello</span>
    </div>;
};
export const Box4 = function({ children, containerClassName }: {
	children?: React.ReactNode;
	containerClassName?: string;
}) {
	return <div className={clsx("pIgHDW_box", containerClassName)}>
      {children}
      <span className="title">Hello</span>
    </div>;
};
export const Box5 = forwardRef<HTMLDivElement, {
	children?: React.ReactNode;
	containerClassName?: string;
}>(({ children, containerClassName }, ref) => {
	return <div className={clsx("f0pFNq_box", containerClassName)} ref={ref}>
        {children}
        <span className="title">Hello</span>
      </div>;
});
export const Box6 = forwardRef<HTMLDivElement, {
	children?: React.ReactNode;
	containerClassName?: string;
}>(({ children, containerClassName }, ref) => {
	return <div className={clsx("qew9fq_box", containerClassName)} ref={ref}>
        {children}
        <span className="title">Hello</span>
        
      </div>;
});
"
`;

exports[`Misc tests > global flair styles are working 1`] = `
"import "@flairjs/client/generated-css/2287b1376d1a3420.css";
import { flair } from "@flairjs/client";
import clsx from "clsx";
export const Box = ({ children, containerClassName }: {
	children?: React.ReactNode;
	containerClassName?: string;
}) => {
	return <div className={clsx("_1fHoyq_box", containerClassName)}>
      {children}
      <span className="title">Hello</span>
    </div>;
};
"
`;

exports[`Misc tests > global styles are working 1`] = `
"import "@flairjs/client/generated-css/7a9cb3081991286f.css";
import { Style } from "@flairjs/client/react";
import clsx from "clsx";
export const Box = ({ children, containerClassName }: {
	children?: React.ReactNode;
	containerClassName?: string;
}) => {
	return <div className={clsx("YS16zq_box", containerClassName)}>
      {children}
      <span className="title">Hello</span>
      
      
    </div>;
};
"
`;
